lib/board.rb

```diff
@@ -0,0 +1,55 @@
+class Board
+  DIRECTIONS = [:NORTH, :EAST, :SOUTH, :WEST]
+
+  attr_accessor :x_pos
+  attr_accessor :y_pos
+  attr_accessor :f_index
+
+  def initialize(x_size, y_size)
+    @x_size = x_size
+    @y_size = y_size
+  end
+
+  def place(x_pos, y_pos, f)
+    x_pos = validate_pos(x_pos, @x_size)
+    y_pos = validate_pos(y_pos, @y_size)
+    f_index = validate_f(f)
+
+    @x_pos = x_pos
+    @y_pos = y_pos
+    @f_index = f_index
+
+    @initialized = true
+  end
+
+  def initialized?
+    @initialized
+  end
+
+  def direction()
+    DIRECTIONS[@f_index]
+  end
+
+  def to_str()
+    "%d,%d,%s" % [@x_pos, @y_pos, DIRECTIONS[@f_index]]
+  end
+
+
+  private
+  def validate_pos(pos, size)
+    pos_int = Integer pos
```

Comment:
why? just use pos. also, name it "position" instead - in Ruby variable names don't take more memory if they're longer, and I highly recommend using meaningful names, as the next reader might have a different abbreviation in mind when they see "pos"

that said, agreed upon abbreviations are okay (i for index, etc)